.github/workflows/build+test.yml

```
@@ -0,0 +1,34 @@
+# Verification workflow to run on all pushed commits.
+# Performs the following checks:
+# - 'yarn install' performs no changes to 'yarn.lock'.
+# - Check formatting in all workspaces.
+# - Build all workspaces.
+# - Run all tests.
+
+name: Build and test
+
+on:
+    push:
```

It seems excessive to run the CI on every push. In most of our repo's we only do it on main and in PR's
Like this is how we have it in the js sdk:
```suggestion
on:
    # Triggers the workflow on push or pull request events but only for the main branch
    push:
        branches: [main, release**]
    pull_request:
        branches: [main, release**]
        # Don't run on draft PR's, see: https://github.com/orgs/community/discussions/25722#discussioncomment-3248917
        types: [opened, synchronize, reopened, ready_for_review]
```

---

I know this is common but I prefer it to be run always. I never push code that I don't want to be checked. The only reason I would want to limit it is if the workflow is computationally heavy. This workflow is light weight: dependencies are cached and it only takes about 3 mins to run.

---

I think it is excessive to run on every push as well. I don't generally check the pipeline runs for every push I do, and I do push code that I don't want to be checked (end-of-day work in progress) as I know it _will_ fail.

I prefer to just keep it aligned with what we do elsewhere so that people don't have to adjust across repositories.

---

There's no harm in having failed workflows. For WIP, they're just signals that you can decide to care about or not.

As a compromise, I updated the trigger to pushes to `main` and all PRs at that's really all I care about. We don't use release branches so I left that part out. Only triggering on PRs targeting specific branches seems like a mistake to me - surely you shouldn't merge broken code anywhere?

I find it annoying when checks are not run against draft PRs, because that means that when I mark the PR as ready and assigns it to reviewers, the first thing they might see is a workflow failure and then more commits to fix that because the PR wasn't actually ready. I prefer to do those fixes before I start bothering them :)